runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/schema/SchemaHandler.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2021-2023 Aklivity Inc.
+ *
+ * Aklivity licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.aklivity.zilla.runtime.engine.schema;
+
+public interface SchemaHandler
+{
+    void configure();
+
+    void schemaRegistryClient();
+
+    int registerSchema(String schemaString, String subject, int version, int id);
+
+    <T> T getSchemasBySubjectAndId(String subject, int id);
+
+    <T> T getSchemasById(int id);
```

Comment:
I think we want to simplify this a bit.

`configure()` seems unnecessary, as we have `attach` and `detach` lifecycle on `SchemaContext` to help with any initialization or configuration based on parsed `options` or even implementation-specific `Configuration`.

`schemaRegistryClient()` seems unnecessary too.

The general mapping we discussed was `schemaId` => some kind of "schema description" that could be used to by caller to validate, encode or decode.

We also discussed a mapping from `schema name` (a.k.a `subject name`) plus optional `version` to `schemaId`.

`registerSchema` could probably be renamed to `register` (since the context of "schema" is already clear) and simplified; "schema / subject name" + "new schema description" => `schemaId`

Let me know if this is sufficient to handle all the use cases.

How do we want to represent the "schema description" and do we need more than one format?
How would a validator / encoder / decoder make use of such a "schema description"?